src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509SubjectKeyIdentifierExtension.cs

```diff
@@ -145,11 +146,35 @@ private static byte[] GenerateSubjectKeyIdentifierFromPublicKey(PublicKey key, X
                     }
 
                 case X509SubjectKeyIdentifierHashAlgorithm.CapiSha1:
-                    return X509Pal.Instance.ComputeCapiSha1OfPublicKey(key);
-
+                    // CAPI SHA1 is the SHA-1 hash over the whole SubjectPublicKeyInfo
+                    return HashSubjectPublicKeyInfo(key, HashAlgorithmName.SHA1);
                 default:
                     throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, algorithm), nameof(algorithm));
             }
         }
+
+        private static byte[] HashSubjectPublicKeyInfo(PublicKey key, HashAlgorithmName hashAlgorithmName)
```

Comment:
This is a _little_ forward thinking because it accepts a `HashAlgorithmName` even though we are only ever passing `SHA1` to it. I am anticipating needing it for https://github.com/dotnet/runtime/issues/97158.

Reply:
Tested here:

https://github.com/dotnet/runtime/blob/70db6299c12684afc7f29efc4465fdef91cb4541/src/libraries/System.Security.Cryptography/tests/X509Certificates/ExtensionsTests/SubjectKeyIdentifierTests.cs#L128